docker/docker.ts

```diff
@@ -545,15 +554,22 @@ async function runCommandThatCanFail(
         });
 
         p.on("error", err => {
-            // If we actually full on received some sort of error in the process, also try to dump
-            // any stderr messages in case they might be useful.
-            logStdErrMessages(/*code: */ 1);
-            reject(err);
+            stdErrChunks.push(new Buffer(err.message));
+            finish(/*code: */ 1);
         });
 
         p.on("close", code => {
-            logStdErrMessages(code);
+            finish(code);
+        });
+
+        if (stdin) {
+            p.stdin.end(stdin);
+        }
+
+        return;
```

Comment:
It makes me nervous that we are touching this code - it has historically had very subtle issues.  Did we have to for this change?